validator_client/src/validator_store.rs

```diff
@@ -60,10 +62,13 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
                     .voting_keypair
                     .clone()
                     .map(|voting_keypair| (voting_keypair.pk, validator_directory))
-            });
+            })
+            .collect::<Vec<_>>();
 
         Ok(Self {
-            validators: Arc::new(RwLock::new(HashMap::from_iter(validator_iter))),
+            validators: Arc::new(RwLock::new(HashMap::from_iter(
+                validator_key_values.into_iter(),
```

Comment:
I think the `.into_iter` is unnecessary, and we might even be able to save collecting into the intermediate Vec by collecting straight into the HashMap: https://docs.rs/rayon/1.3.0/rayon/iter/trait.FromParallelIterator.html

Reply:
Good point about `into_iter`.

I tried getting rid of the collect earlier, I can't just collect into the Hashmap because the compiler doesn't recognize the return of the `filter_map` as an iterator.

[hashbrown](https://docs.rs/hashbrown/0.6.0/hashbrown/hash_map/rayon/index.html) has some support for collecting from Rayon into a HashMap but AFAICT `std::collections::HashMap` didn't carry that over.

Reply:
`HashMap` implements `FromParallelIterator`, this seems to compile OK:

```diff
diff --git a/validator_client/src/validator_store.rs b/validator_client/src/validator_store.rs
index 62696293..ac7ebef8 100644
--- a/validator_client/src/validator_store.rs
+++ b/validator_client/src/validator_store.rs
@@ -62,11 +62,10 @@ impl<T: SlotClock + 'static, E: EthSpec> ValidatorStore<T, E> {
                     .voting_keypair
                     .clone()
                     .map(|voting_keypair| (voting_keypair.pk, validator_directory))
-            })
-            .collect::<Vec<_>>();
+            });
 
         Ok(Self {
-            validators: Arc::new(RwLock::new(HashMap::from_iter(validator_key_values))),
+            validators: Arc::new(RwLock::new(HashMap::from_par_iter(validator_key_values))),
             spec: Arc::new(spec),
             log,
             temp_dir: None,
```